### PR TITLE
Update conda to clang 11.1.0

### PR DESCRIPTION
**Pull request by cticenhour**
Updates conda clang to 11.1.0 in order to restore compatibility with Xcode / CLT 12.5.

Refs #16291

### CODE REVIEW

File: conda/mpich/conda_build_config.yaml [hunk 1]
```diff
@@ -14,29 +14,29 @@ vtk_version:
...
-vtk:
-  - 9.0.1 no_osmesa_py39h3dd132e_102 # [osx]
-  - 9.0.1 no_osmesa_py38h523a0c9_102 # [osx]
-  - 9.0.1 no_osmesa_py37hc0470e1_102 # [osx]
-  - 9.0.1 no_osmesa_py36hc362858_102 # [osx]
+moose_vtk:
+  - 9.0.1 no_osmesa_py39h9af63fb_109 # [osx]
+  - 9.0.1 no_osmesa_py38h1d02333_109 # [osx]
+  - 9.0.1 no_osmesa_py37hbb2da5a_109 # [osx]
+  - 9.0.1 no_osmesa_py36h6f68f9a_109 # [osx]
```
Comment (milljm) [inline conda/mpich/conda_build_config.yaml:33]: Well done figuring these out!

**Review by milljm — DISMISSED** [review]
I am still building manually. But these look good. If successful, merge away.

**Comment by cticenhour** [comment]
@milljm here's a bit of a long message summarizing the lay of the land. It's good to see that we've been able to get the full complement of packages built! But it looks like that issue with linking is a big remaining problem, looking at the most recent results (https://civet.inl.gov/job/787074/). The dynamic libraries don't seem to be built correctly. Worryingly, I've discovered that I am also having issues building PETSc manually on a Big Sur workstation with Xcode 12.4 (wanted to check against backwards compatibility since I've been building with 12.5) when using the `conda.software.inl.gov` version of `moose-mpich`. It fails at finding Fortran:
```
TESTING: consistencyChecks from config.packages.fblaslapack(config/BuildSystem/config/package.py:963)                                                                                                                                                          *******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
Cannot use fblaslapack without Fortran, make sure you do NOT have --with-fc=0
*******************************************************************************
```
The `gfortran` package is loaded and the Fortran build flags in my environment all appear to be there, with `which mpif90` directing me to the expected location in my conda environment `bin` directory. In the `configure.log` I'm seeing:
```
Executing: mpif90 -c -o /var/folders/4h/557m8xtx2w9b6yxdhzy3b2gr0000gn/T/petsc-9oh8ikzq/config.setCompilers/conftest.o -I/var/folders/4h/557m8xtx2w9b6yxdhzy3b2gr0000gn/T/petsc-9oh8ikzq/config.setCompilers   /var/folders/4h/557m8xtx2w9b6yxdhzy3b2gr0000gn/T/petsc-9oh8ikzq/config.setCompilers/conftest.F90 
Successful compile:
Source:
      program main

      end
Executing: mpif90  -o /var/folders/4h/557m8xtx2w9b6yxdhzy3b2gr0000gn/T/petsc-9oh8ikzq/config.setCompilers/conftest    /var/folders/4h/557m8xtx2w9b6yxdhzy3b2gr0000gn/T/petsc-9oh8ikzq/config.setCompilers/conftest.o 
Possible ERROR while running linker: exit code 1
stderr:
ld: library not found for -lSystem
collect2: error: ld returned 1 exit status
Linker output before filtering:

ld: library not found for -lSystem
collect2: error: ld returned 1 exit status
:
Linker output after filtering:

ld: library not found for -lSystem
collect2: error: ld returned 1 exit status:
          Error testing Fortran compiler: Cannot compile/link FC with mpif90.
           MPI installation mpif90 is likely incorrect.
```
Which again suggests that the linker is the issue, though it was able to successfully compile the first `conftest` file. Another final puzzler, I can build `moose-petsc` locally just fine, so it's finding Fortran there when `moose-mpich` is loaded in the build environment....

Two options for changing `ld64` exist - retaining the `530` version and going to a newer build (only one more past the one we're using in this PR), or going to version `609`, which seems to require a bump to clang `12.0.1`, as mamba cannot resolve it with `11.1.0`. I'll continue to work on this this afternoon.

**Comment by cticenhour** [comment]
Closing in favor of an upgrade to clang 12, to coincide with Mac M1 processor support.

**Pull request closed without merging**